KinectSaveModel/SkeletonReady.cs

```diff
@@ -0,0 +1,176 @@
+
+            // create frame from the writable bitmap and add to encoder
+            encoder.Frames.Add(BitmapFrame.Create(MainWindow.colorBitmap));
+
+            string time = System.DateTime.Now.ToString("hh'-'mm'-'ss", CultureInfo.CurrentUICulture.DateTimeFormat);
+            if (frameNumber == 0)
+            {
+                if (!Directory.Exists(@"C:\KinectSavedMovements\screenshots\FirstMovement"))
+                {
+                    Directory.CreateDirectory(@"C:\KinectSavedMovements\screenshots\FirstMovement");
+                    ssPath = @"C:\KinectSavedMovements\screenshots\FirstMovement";
+                }
+                if (Directory.Exists(@"C:\KinectSavedMovements\screenshots\FirstMovement"))
+                {
+                    Directory.CreateDirectory(@"C:\KinectSavedMovements\screenshots\FirstMovement-"+time);
+                    ssPath = @"C:\KinectSavedMovements\screenshots\FirstMovement-" + time;
+                }
+            }
+
+            string ssFullPath = Path.Combine(ssPath, "frame-" + frameNumber + ".png");
+
+            // write the new file to disk
+            try
+            {
+                using (FileStream fs = new FileStream(ssFullPath, FileMode.Create))
+                {
+                    encoder.Save(fs);
+                }
+
+                //this.statusBarText.Text = string.Format(CultureInfo.InvariantCulture, "{0} {1}", Properties.Resources.ScreenshotWriteSuccess, path);
+            }
+            catch (IOException)
+            {
+                //this.statusBarText.Text = string.Format(CultureInfo.InvariantCulture, "{0} {1}", Properties.Resources.ScreenshotWriteFailed, path);
+            }
+        }
+
+        private void calculate(Joint rightShoulder, Joint rightElbow, Joint rightWrist)
+        {
+            float rightShoulderCoordx = coordinatePos(640, rightShoulder.Position.X);
+            float rightElbowCoordx = coordinatePos(640, rightElbow.Position.X);
+            float rightShoulderCoordy = coordinatePos(480, rightShoulder.Position.X);
+            float rightElbowCoordy = coordinatePos(480, rightElbow.Position.X);
+            double lengthX = Math.Pow((rightElbowCoordx - rightShoulderCoordx),2);
+            double lengthY = Math.Pow((rightElbowCoordy-rightShoulderCoordy),2);
+            double lengthUpper = Math.Sqrt((lengthX+lengthY));
+            double asin = Math.Sqrt(lengthX) / lengthUpper;
+            //asin = Math.Acos(asin);
+            //MainWindow.main.statusBarText.Text = ""+lengthY;
+            MainWindow.main.statusBarText.Text = ""+Math.Sqrt(lengthX)+"/"+lengthUpper+"="+asin+" = asin("+(Math.Acos(asin)*(180/Math.PI))+"";
+        }
+
+        private void writeToFile(List<Vector3D> vectorList)
+        {
+            string line = "";
+            String timeStamp = GetTimestamp(DateTime.Now);
+            String frame = "" + frameNumber;
+            line += frame + ",[" + timeStamp + "]@";
```

Comment:
deze methode heeft sowieso wat comments nodig maar ik vraag me af of het niet makkelijker kan. Ik kan met niet voorstellen dat dit de aanbevolen manier is om posities weg te schrijven met betrekking tot Kinect.